src/lib/change_cmdgen.c

```diff
@@ -308,10 +312,109 @@ int creat_cmd_key_and_value(struct lyd_node *dnode, oper_t curr_op_val, char **k
     return EXIT_SUCCESS;
 }
 
+/**
+ * create iproute2 arguments out of lyd2 node, this will take the op_val into consideration,
+ * @param startcmd_node lyd_node to generate arg for, example link, nexthop, filter ... etc
+ * @param op_val operation value
+ * @return
+ */
+char *lyd2cmdline_args(const struct lyd_node *startcmd_node, oper_t op_val)
+{
+    char cmd_line[CMD_LINE_SIZE] = {0};
+
+    int ret;
+    struct lyd_node *next;
+    LYD_TREE_DFS_BEGIN(startcmd_node, next)
+    {
+        char *on_update_include = NULL;
+        char *key = NULL, *value = NULL;
+        switch (next->schema->nodetype) {
+            case LYS_LIST:
+            case LYS_CONTAINER:
+                if (op_val == UPDATE_OPR &&
+                    get_extension(ON_UPDATE_INCLUDE, next, &on_update_include) == EXIT_SUCCESS) {
+                    // capture the on-update-include ext,
+                    if (on_update_include == NULL) {
+                        fprintf(stderr, "%s: ON_UPDATE_INCLUDE extension found,"
+                                        "but failed to retrieve the arg-name list from ON_UPDATE_INCLUDE extension for node \"%s\" \n",
+                                __func__, next->schema->name);
+                        return NULL;
+                    }
+                    // get all the on_update_include arg-names, fetch them from sr, and add them to the cmd_line.
+                    // args-names format = arg1, arg2, ... argn
+                    char *token;
+
+                    // Get the first arg
+                    token = strtok(on_update_include, ",");
```

Comment:
can we make this a function ?

Reply:
pull from sysrepo:
input: node name(s)
output: either lyd_node or cmd args list